src/solver/hydro/management/daily.cpp

```diff
@@ -422,7 +422,7 @@ inline void HydroManagement::prepareDailyOptimalGenerations(Solver::Variable::St
                 break;
             }
 
-            H2O_J_Free(&problem);
+            H2O_J_Free(problem.get());
```

Comment:
Careful not to call `delete` or `free` on `problem.get()`, this would lead to a double-free since `~unique_ptr()` also de-allocates the memory.